### PR TITLE
Faster domain filter

### PR DESCRIPTION
**Pull request by hellais**
Query time for an example query has gone from 537,136.721 ms
to 146.392 ms.

This improvement is achieved by creating a CTE and using it to hint the
query planner as to how to do the cross table JOIN.

### CODE REVIEW

File: measurements/api/measurements.py [hunk 1]
```diff
@@ -16,7 +16,8 @@
 from flask.json import jsonify
 from werkzeug.exceptions import HTTPException, BadRequest
 
-from sqlalchemy import func, or_, and_, false, true, text
+from sqlalchemy.dialects import postgresql
+from sqlalchemy import func, or_, and_, false, true, text, select, sql, column
```
Comment (FedericoCeratto) [inline measurements/api/measurements.py]: "true" is unused

File: measurements/database.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 from sqlalchemy import event
 from sqlalchemy.engine import Engine
 
+from sqlalchemy.dialects import postgresql
```
Comment (FedericoCeratto) [inline measurements/database.py]: unused

File: measurements/api/measurements.py [hunk 3]
```diff
@@ -200,31 +201,33 @@ def get_measurement(measurement_id, download=None):
                 'attachment', filename=filename)
     return response
 
-def input_filter(q, input_, domain, test_name):
+def input_cte(input_, domain, test_name):
```
Comment (FedericoCeratto) [inline measurements/api/measurements.py]: input_cte appears not to be covered in https://travis-ci.org/ooni/api/builds/581319654#L2112  - is it worth testing?

Comment (hellais) [reply measurements/api/measurements.py]: I added a basic check to the smoke test. Without fixtures in the DB though it's hard to have a really useful test in here though.

**Pull request merged**